plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -59,34 +334,169 @@ constraints, affinity and anti-affinity specifications, data locality, inter-wor
+			config.LeaderElection.LockObjectNamespace,
+			config.LeaderElection.LockObjectName,
+			leaderElectionClient.CoreV1(),
+			resourcelock.ResourceLockConfig{
+				Identity:      hostname,
+				EventRecorder: recorder,
+			})
+		if err != nil {
+			return nil, fmt.Errorf("couldn't create resource lock: %v", err)
+		}
+
+		leaderElectionConfig = &leaderelection.LeaderElectionConfig{
+			Lock:          rl,
+			LeaseDuration: config.LeaderElection.LeaseDuration.Duration,
+			RenewDeadline: config.LeaderElection.RenewDeadline.Duration,
+			RetryPeriod:   config.LeaderElection.RetryPeriod.Duration,
+		}
+	}
+
+	// Prepare a healthz server. If the metrics bind address is the same as the
+	// healthz bind address, consolidate the servers into one.
+	var healthzServer *http.Server
+	if len(config.HealthzBindAddress) > 0 {
+		mux := http.NewServeMux()
+		healthz.InstallHandler(mux)
+		if config.HealthzBindAddress == config.MetricsBindAddress {
+			configz.InstallHandler(mux)
+			mux.Handle("/metrics", prometheus.Handler())
+		}
+		if config.EnableProfiling {
+			mux.HandleFunc("/debug/pprof/", pprof.Index)
+			mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+			mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+			mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
+			if config.EnableContentionProfiling {
+				goruntime.SetBlockProfileRate(1)
+			}
+		}
+		healthzServer = &http.Server{
+			Addr:    config.HealthzBindAddress,
+			Handler: mux,
+		}
+	}
+
+	// Prepare a separate metrics server only if the bind address differs from the
+	// healthz bind address.
+	var metricsServer *http.Server
+	if len(config.MetricsBindAddress) > 0 && config.HealthzBindAddress != config.MetricsBindAddress {
+		mux := http.NewServeMux()
+		configz.InstallHandler(mux)
+		mux.Handle("/metrics", prometheus.Handler())
+		if config.EnableProfiling {
+			mux.HandleFunc("/debug/pprof/", pprof.Index)
+			mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+			mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+			mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
+			if config.EnableContentionProfiling {
+				goruntime.SetBlockProfileRate(1)
+			}
+		}
+		metricsServer = &http.Server{
+			Addr:    config.MetricsBindAddress,
+			Handler: mux,
+		}
+	}
+
+	return &SchedulerServer{
+		SchedulerName:                  config.SchedulerName,
+		Client:                         client,
+		AlgorithmSource:                config.AlgorithmSource,
+		HardPodAffinitySymmetricWeight: config.HardPodAffinitySymmetricWeight,
+		EventClient:                    eventClient,
+		Recorder:                       recorder,
+		Broadcaster:                    eventBroadcaster,
+		LeaderElection:                 leaderElectionConfig,
+		HealthzServer:                  healthzServer,
+		MetricsServer:                  metricsServer,
+	}, nil
+}
 
-	sched, err := CreateScheduler(
-		s,
-		kubeClient,
+// Run runs the SchedulerServer. This should never exit.
+func (s *SchedulerServer) Run() error {
```

Comment:
Same as above. Please break this into smaller functions.

Reply:
Decomposed.